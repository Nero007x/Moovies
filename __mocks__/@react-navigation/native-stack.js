export const createNativeStackNavigator = jest.fn(() => ({
    Navigator: "StackNavigator",
    Screen: "Screen",
  }));
  