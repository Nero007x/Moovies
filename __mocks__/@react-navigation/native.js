export const useNavigation = () => ({
    navigate: jest.fn(),
  });
  
  export const NavigationContainer = ({ children }) => children;
  