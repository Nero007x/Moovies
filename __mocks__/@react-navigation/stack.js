export const createStackNavigator = jest.fn(() => ({
    Navigator: "StackNavigator",
    Screen: "Screen",
}));
