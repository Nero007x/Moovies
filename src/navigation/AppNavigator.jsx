import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import HomeScreen from '../screens/HomeScreen';
import MovieDetails from '../screens/MovieDetails';

const Stack = createStackNavigator();

const AppNavigator = () => (
  <Stack.Navigator>
    <Stack.Screen name="Home" component={HomeScreen} />
    <Stack.Screen name="MovieDetails" component={MovieDetails} />
  </Stack.Navigator>
);

export default AppNavigator;
