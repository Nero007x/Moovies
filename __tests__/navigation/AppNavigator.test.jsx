import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import AppNavigator from '../../src/navigation/AppNavigator';

jest.mock('../../src/screens/HomeScreen', () => 'HomeScreen');
jest.mock('../../src/screens/MovieDetails', () => 'MovieDetails');

describe('AppNavigator', () => {
  it('matches the snapshot', () => {
    const tree = render(
      <NavigationContainer>
        <AppNavigator />
      </NavigationContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
