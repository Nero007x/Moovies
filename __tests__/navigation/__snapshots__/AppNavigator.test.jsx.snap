// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavigator matches the snapshot 1`] = `
<StackNavigator>
  <Screen
    component="HomeScreen"
    name="Home"
  />
  <Screen
    component="MovieDetails"
    name="MovieDetails"
  />
</StackNavigator>
`;
